### ISSUE TITLE
Default message for unskilled Bot

### ISSUE THREAD

**Issue by daniccan** [opsdroid/opsdroid#1263]
# Description
The Bot currently does not respond to messages that it is not skilled to answer. It would be good to have a default message/matcher set to the Bot so that it can respond with the default message if it is not skilled for that particular scenario.

An example default message would be,
`I’m sorry, but I don’t know that.` or `I’m sorry, but I don’t understand that.`

## Steps to Reproduce
1. Create a Bot in Opsdroid with `hello` skill.
2. Ask the Bot `How are you?`.

## Expected Functionality
The Bot replies with the message `I’m sorry, but I don’t understand that.`.

## Experienced Functionality
The Bot does not respond.

## Versions
- **Opsdroid version:** master branch in git
- **Python version:** 3.6.8
- **OS/Docker version:** Ubuntu 18.04

**Comment by daniccan** [comment]
I was thinking of adding an optional configuration `default-bot-reply` which can be used as response of the Bot if there are no matches.

**Comment by jacobtomlinson** [comment]
Thanks for raising this. I would be happy to add functionality like this but we should be thoughtful about how we implement it.

Here are a few things to consider:
- Should this be a matcher, or a configuration option?
- If multiple skills are installed which should be giving the default response?
- ChatOps bots are often in group rooms where many people will be talking. We probably don't want the bot to respond to everything in that case. 
  - Should we try and detect this and only give the default response if the message was directed at the bot (would depend on #273)? 
  - Should we just leave this as a configurable option?
- Getting the same error repeatedly would be frustrating. 
  - Do we want a list of variations to randomly choose from?
  - Do we want some kind of intelligent help similar to [thefuck](https://github.com/nvbn/thefuck)?

Discussion and input from others is very welcome!

**Comment by FabioRosado** [comment]
I'm finding a problem when using opsdroid in a room full of people because opsdroid will trigger on every single message and spam the room. Perhaps this issue should be implemented together with #273? 

So when opsdroid is indeed addressed to we should give a default message if we can't understand the message given. 

I agree that we should give a list of potential replies to give the user, perhaps setting them on the config.yaml which allows users to specify a list of possible answers something like:

```python
default-message:
  - Sorry, I couldn't understand that.
  - Unable to process command.
  - ...
```

**Comment by jacobtomlinson** [comment]
Agreed. I think we could implement it without #273 and just set the default to

```yaml
default-message: []
```

which just disables it anyway. Then folks who know the bot will only ever be spoken to directly could configure it.

Then we could add #273 later as an extra feature which then makes it even better!
